src/storage/src/storage_state.rs

```diff
@@ -1072,43 +1085,38 @@ impl<'w, A: Allocate> Worker<'w, A> {
                             let running_ingestion =
                                 self.storage_state.ingestions.get(&ingestion.id);
 
-                            // Ingestion statements are only considered updates if they are
-                            // currently running.
-                            ingestion.update = running_ingestion.is_some();
-
                             // We keep only:
                             // - The most recent version of the ingestion, which
                             //   is why these commands are run in reverse.
                             // - Ingestions whose descriptions are not exactly
                             //   those that are currently running.
-                            seen_most_recent_ingestion.insert(ingestion.id)
+                            seen_most_recent_definition.insert(ingestion.id)
                                 && running_ingestion != Some(&ingestion.description)
                         }
                     })
                 }
-                StorageCommand::CreateSinks(exports) => {
+                StorageCommand::RunSinks(exports) => {
                     exports.retain_mut(|export| {
-                        if drop_commands.remove(&export.id) {
+                        if drop_commands.remove(&export.id)
+                            || self.storage_state.dropped_ids.contains(&export.id)
```

Comment:
can we add a comment explaining what scenario causes us to not have a drop command but to continue to have `dropped_ids`?

Reply:
Yeah––this could happen if there were multiple `RunSink` commands in the stream in addition to the drop––i.e. the drop command processing only catches the first instance.